brushfire-training/src/main/scala/com/stripe/brushfire/local/CrossEntropyExample.scala

```diff
@@ -0,0 +1,44 @@
+package com.stripe.brushfire.local
+
+import com.stripe.bonsai._
+import com.stripe.brushfire._
+import com.twitter.algebird._
+
+import AnnotatedTree.{AnnotatedTreeTraversal, fullBinaryTreeOpsForAnnotatedTree}
+
+object CrossEntropyExample extends Defaults {
+
+  def main(args: Array[String]) {
+    val path = args.head
+    val cols = args.tail.toList
+
+    // T = Map[String, Long]
+    // P = String
+    // E = Double
+
+    val trainingData = Lines(path).map { line =>
+      val parts = line.split(",").reverse.toList
+      val label = parts.head
+      val values = parts.tail.map { s => s.toDouble }
+      Instance(line, 0L, Map(cols.zip(values): _*), Map(label -> 1L))
+    }.toIterable
+
+    val totalData: Map[String, Long] = Monoid.sum(trainingData.iterator.map(_.target))
```

Comment:
It seems wrong to me that we are using the full training set, in any way, in an error computation; I feel like we should only be basing the error computation on the validation set.